# Exploring and cleaning the data

## EDA

In [3]:
import pandas as pd
import numpy as np
import warnings

warnings.filterwarnings('ignore')

In [4]:
shelter_statistics_df = pd.read_csv('../data/animal-care-and-control-data-set-2004-2020.csv')
pet_df = pd.read_csv('../data/Animal_Shelter_Animals.csv')

In [5]:
pet_df.head()

id              intakedate intakereason  istransfer sheltercode  \
0  15801  11/28/2009 12:00:00 AM       Moving       False   C09115463   
1  15932  12/08/2009 12:00:00 AM       Moving       False   D09125594   
2  28859  08/10/2012 12:00:00 AM    Abandoned       False   D12082309   
3  28859  08/10/2012 12:00:00 AM    Abandoned       False   D12082309   
4  30812  01/11/2013 12:00:00 AM    Abandoned       False    C1301091   

  identichipnumber   animalname                breedname  \
0       0A115D7358       Jadzia      Domestic Short Hair   
1       0A11675477        Gonzo  German Shepherd Dog/Mix   
2       0A13253C7B       Maggie  Shep Mix/Siberian Husky   
3       0A13253C7B       Maggie  Shep Mix/Siberian Husky   
4       0A13403D4D  Pretty Girl      Domestic Short Hair   

             basecolour speciesname  ...            movementdate movementtype  \
0         Tortoiseshell         Cat  ...  05/13/2017 12:00:00 AM     Adoption   
1                   Tan         Dog  ...  04/24/2017 12:00:00 AM     Adoption   
2               Various         Dog  ...  04/15/2017 12:00:00 AM     Adoption   
3               Various         Dog  ...  02/29/2020 12:00:00 AM    Reclaimed   
4  Dilute tortoiseshell         Cat  ...  04/18/2017 12:00:00 AM       Foster   

  istrial              returndate returnedreason  deceaseddate deceasedreason  \
0   False                     NaN          Stray           NaN   Died in care   
1   False                     NaN          Stray           NaN   Died in care   
2   False  02/28/2020 12:00:00 AM          Stray           NaN   Died in care   
3   False                     NaN          Stray           NaN   Died in care   
4   False  05/29/2018 12:00:00 AM          Stray           NaN   Died in care   

  diedoffshelter puttosleep  isdoa  
0          False      False  False  
1          False      False  False  
2          False      False  False  
3          False      False  False  
4          False      False  False  

[5 rows x 23 columns]

In [6]:
pet_df.tail(10)

id              intakedate              intakereason  istransfer  \
18669  71693  11/01/2021 02:11:10 PM             Police Assist       False   
18670  71694  11/01/2021 02:11:10 PM             Police Assist       False   
18671  71695  11/01/2021 02:11:10 PM             Police Assist       False   
18672  71696  11/01/2021 02:11:10 PM             Police Assist       False   
18673  71697  11/01/2021 02:11:10 PM             Police Assist       False   
18674  71705  11/02/2021 04:40:39 PM  Unsuitable Accommodation       False   
18675  71728  11/05/2021 12:02:28 PM                     Stray       False   
18676  71730  11/05/2021 01:03:56 PM                     Stray       False   
18677  71753  11/06/2021 03:04:53 PM                     Stray       False   
18678  71756  11/07/2021 01:26:44 PM                     Stray       False   

      sheltercode identichipnumber animalname  \
18669   C21111474              NaN      Kit 1   
18670   C21111475              NaN      Kit 2   
18671   C21111476              NaN      Kit 3   
18672   C21111477              NaN      Kit 4   
18673   C21111478              NaN      Kit 5   
18674   D21111458              NaN      Daisy   
18675   C21111491  981020045195439       Coco   
18676   D21111493  956000010131703      Honey   
18677   C21111507  981020023519919        Max   
18678   C21111508              NaN     Willow   

                               breedname       basecolour speciesname  ...  \
18669                 Domestic Long Hair  Brown and Black         Cat  ...   
18670                 Domestic Long Hair  Brown and Black         Cat  ...   
18671                 Domestic Long Hair  Brown and Black         Cat  ...   
18672                 Domestic Long Hair  Brown and Black         Cat  ...   
18673                 Domestic Long Hair  Brown and Black         Cat  ...   
18674                Pug/Bulldog, French            Brown         Dog  ...   
18675                Domestic Short Hair        Chocolate         Cat  ...   
18676  Golden Retriever/Poodle, Standard           Golden         Dog  ...   
18677                Domestic Short Hair             Grey         Cat  ...   
18678                Domestic Short Hair            Black         Cat  ...   

                 movementdate movementtype istrial returndate returnedreason  \
18669  11/02/2021 12:00:00 AM    Reclaimed   False        NaN          Stray   
18670  11/02/2021 12:00:00 AM    Reclaimed   False        NaN          Stray   
18671  11/02/2021 12:00:00 AM    Reclaimed   False        NaN          Stray   
18672  11/02/2021 12:00:00 AM    Reclaimed   False        NaN          Stray   
18673  11/02/2021 12:00:00 AM    Reclaimed   False        NaN          Stray   
18674  11/03/2021 12:00:00 AM       Foster   False        NaN          Stray   
18675  11/06/2021 12:00:00 AM    Reclaimed   False        NaN          Stray   
18676  11/05/2021 12:00:00 AM    Reclaimed   False        NaN          Stray   
18677  11/07/2021 12:00:00 AM    Reclaimed   False        NaN          Stray   
18678  11/07/2021 12:00:00 AM       Foster   False        NaN          Stray   

       deceaseddate      deceasedreason diedoffshelter puttosleep  isdoa  
18669           NaN  Court Order/ Legal          False      False  False  
18670           NaN  Court Order/ Legal          False      False  False  
18671           NaN  Court Order/ Legal          False      False  False  
18672           NaN  Court Order/ Legal          False      False  False  
18673           NaN  Court Order/ Legal          False      False  False  
18674           NaN  Court Order/ Legal          False      False  False  
18675           NaN  Court Order/ Legal          False      False  False  
18676           NaN  Court Order/ Legal          False      False  False  
18677           NaN  Court Order/ Legal          False      False  False  
18678           NaN  Court Order/ Legal          False      False  False  

[10 rows x 23 columns]

In [7]:
pet_df.columns

Index(['id', 'intakedate', 'intakereason', 'istransfer', 'sheltercode',
       'identichipnumber', 'animalname', 'breedname', 'basecolour',
       'speciesname', 'animalage', 'sexname', 'location', 'movementdate',
       'movementtype', 'istrial', 'returndate', 'returnedreason',
       'deceaseddate', 'deceasedreason', 'diedoffshelter', 'puttosleep',
       'isdoa'],
      dtype='object')

In [8]:
pet_df.shape

(18679, 23)

In [9]:
pet_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18679 entries, 0 to 18678
Data columns (total 23 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   id                18679 non-null  int64 
 1   intakedate        18679 non-null  object
 2   intakereason      18677 non-null  object
 3   istransfer        18679 non-null  bool  
 4   sheltercode       18679 non-null  object
 5   identichipnumber  15639 non-null  object
 6   animalname        18679 non-null  object
 7   breedname         18626 non-null  object
 8   basecolour        18679 non-null  object
 9   speciesname       18679 non-null  object
 10  animalage         18679 non-null  object
 11  sexname           18679 non-null  object
 12  location          18679 non-null  object
 13  movementdate      18629 non-null  object
 14  movementtype      18679 non-null  object
 15  istrial           18679 non-null  bool  
 16  returndate        6429 non-null   object
 17  returnedreas

- Columns 18 through 22 are results, so they will not be used in this type of classification
- Features and target are currently strings
- There are not have many missing values in the columns that will be used in modeling
- deceased date is mostly nan but is not a column that will be used

In [10]:
pet_df['sheltercode'].value_counts()

D18081107    11
D1801048     11
C18101445    10
D18111619    10
D2102130      8
             ..
C2102119      1
D21101400     1
C1707791      1
D17031735     1
C1901027      1
Name: sheltercode, Length: 12513, dtype: int64

There are 12513 shelter codes, and the most frequent code only has 11 so this column doesnt tell us much 

In [11]:
pet_df['breedname'].value_counts()

Domestic Short Hair           9240
Domestic Medium Hair           861
Domestic Long Hair             617
Bully Breed Mix                616
Labrador Retriever             258
                              ... 
Bengal/Domestic Short Hair       1
Golden Retriever/Sheltie         1
Retriever/Beagle                 1
Pixie-Bob                        1
Vizsla                           1
Name: breedname, Length: 1065, dtype: int64

In [12]:
pet_df['basecolour'].value_counts()

Black              2727
Black and White    1636
Grey               1150
Black and Brown    1138
Orange              975
                   ... 
Tabby and White       5
Tortie Point          4
Tabby                 4
Liver                 3
White and Torti       1
Name: basecolour, Length: 80, dtype: int64

- 1065 different breednames
- 80 different colors
- breenames will need to be binned, I do not want 1000 columns in my modeling dataset

In [13]:
pet_df['istrial'].value_counts()

False    18679
Name: istrial, dtype: int64

This columns is 100% false which makes it useless for classification purposes

In [14]:
pet_df['returnedreason'].value_counts()

Stray                                17366
Incompatible with owner lifestyle      417
Return Adopt - Behavior                164
Return adopt - lifestyle issue         154
Moving                                  81
Return Adopt - Other                    69
Incompatible with other pets            53
Unsuitable Accommodation                50
Biting                                  42
Allergies                               40
Landlord issues                         36
Unable to Afford                        33
Owner requested Euthanasia              33
Rabies Monitoring                       22
Police Assist                           21
Behavioral Issues                       20
Return Adopt - Animal Health            18
DOA                                     13
Sick/Injured                            13
Abandoned                               11
Transfer from Other Shelter              9
Owner Deceased                           9
Marriage/Relationship split              4
Abuse/ negl

95 percent of the returned reason is 'stray', so it wont make much of a difference in a classification model

In [15]:
pet_df['istransfer'].value_counts()

False    18606
True        73
Name: istransfer, dtype: int64

99 percent False, not very useful. This column will be dropped 

In [16]:
pet_df['puttosleep'].value_counts()

False    18386
True       293
Name: puttosleep, dtype: int64

In [17]:
pet_df['deceasedreason'].value_counts()

Court Order/ Legal                       17566
Died in care                               772
UU - untreatable, unmanageable health      149
Owner Requested                             37
UU-untreatable, unmanageable Behavior       37
Dead On Arrival                             34
Biting                                      20
Medical                                     16
Temperament/Behavior                        12
TM - Treatable Manageable Behavior          12
Vet advised euthanasia                       9
Died in community                            8
Sick/Injured                                 6
Healthy                                      1
Name: deceasedreason, dtype: int64

The reason that a dog died is not really a feature in prediciting adoption speed.

In [18]:
pet_df['intakereason'].value_counts() 

Stray                                9042
Incompatible with owner lifestyle    2026
Litter relinquishment                2006
Moving                                757
Unsuitable Accommodation              712
Unable to Afford                      622
Abandoned                             561
Born in Shelter                       518
Police Assist                         413
Transfer from Other Shelter           405
Landlord issues                       301
Sick/Injured                          222
Owner Deceased                        214
Allergies                             204
Incompatible with other pets          149
TNR - Trap/Neuter/Release             132
Marriage/Relationship split            91
Biting                                 85
Rabies Monitoring                      81
Behavioral Issues                      41
Abuse/ neglect                         36
Owner Died                             27
Owner requested Euthanasia             20
Injured Wildlife                  

### Cleaning/Feature Engineering

#### Fix datetime columns 

Dates are currently strings so we will change them to datetime objects

In [19]:
pet_df['movementdate'] = pd.to_datetime(pet_df['movementdate'])
pet_df['intakedate'] = pd.to_datetime(pet_df['intakedate'])

Drop columns that are not useful for predicting

In [20]:
pet_df.drop(columns= ['sheltercode', 'isdoa', 'diedoffshelter', 'deceasedreason', 'deceaseddate', 'returnedreason', 'istrial'], inplace=True)

Intake reason can be a feature, will bin smaller values into a value called 'other'.

In [21]:
def bin_intakereason(row):
    '''keep the top 6 values and bin the rest into a value called other
    '''
    if row['intakereason'] == 'Stray':
        return 'Stray'
    elif row['intakereason'] == 'Incompatible with owner lifestyle':
        return 'Incompatible'
    elif row['intakereason'] == 'Litter relinquishment':
        return 'Litter relinquishment'
    elif row['intakereason'] == 'Moving':
        return 'Moving'
    elif row['intakereason'] == 'Unsuitable Accommodation':
        return 'Unsuitable Accommodation'
    elif row['intakereason'] == 'Unable to Afford':
        return 'Unable to Afford'
    elif row['intakereason'] == 'Abandoned':
        return 'Abandoned'
    else: return 'other'
    
pet_df['intakereason'] = pet_df.apply(lambda row: bin_intakereason(row), axis=1)

In [22]:
pet_df['intakereason'].value_counts()

Stray                       9042
other                       2953
Incompatible                2026
Litter relinquishment       2006
Moving                       757
Unsuitable Accommodation     712
Unable to Afford             622
Abandoned                    561
Name: intakereason, dtype: int64

In [23]:
pet_df.head()

id intakedate intakereason  istransfer identichipnumber   animalname  \
0  15801 2009-11-28       Moving       False       0A115D7358       Jadzia   
1  15932 2009-12-08       Moving       False       0A11675477        Gonzo   
2  28859 2012-08-10    Abandoned       False       0A13253C7B       Maggie   
3  28859 2012-08-10    Abandoned       False       0A13253C7B       Maggie   
4  30812 2013-01-11    Abandoned       False       0A13403D4D  Pretty Girl   

                 breedname            basecolour speciesname  \
0      Domestic Short Hair         Tortoiseshell         Cat   
1  German Shepherd Dog/Mix                   Tan         Dog   
2  Shep Mix/Siberian Husky               Various         Dog   
3  Shep Mix/Siberian Husky               Various         Dog   
4      Domestic Short Hair  Dilute tortoiseshell         Cat   

            animalage sexname                    location movementdate  \
0  11 years 8 months.  Female  Adoptable Cat Glass Colony   2017-05-13   
1  10 years 9 months.    Male              Adoptable Dogs   2017-04-24   
2    8 years 1 month.  Female          Canine intake room   2017-04-15   
3    8 years 1 month.  Female          Canine intake room   2020-02-29   
4  10 years 6 months.  Female                  Stray Cats   2017-04-18   

  movementtype              returndate  puttosleep  
0     Adoption                     NaN       False  
1     Adoption                     NaN       False  
2     Adoption  02/28/2020 12:00:00 AM       False  
3    Reclaimed                     NaN       False  
4       Foster  05/29/2018 12:00:00 AM       False

In [24]:
pet_df['location'].value_counts()

Adoptable Dogs                                                          3210
Adoptable window colony                                                 1675
Adoptable Cat Kennels                                                   1649
Adoptable Cat Glass Colony                                              1148
Adoptable Cat Middle Colony                                             1051
Adoptable Cat Big Colony                                                 989
Adoptable Cat Small Colony                                               972
Lobby                                                                    879
Cat room A                                                               758
Canine intake room                                                       647
Dog room A                                                               588
Cat room C                                                               520
Dog room C                                                               438

The location column tells the animal's location inside of the shelter, so thats not really useful for prediciton purposes.

In [25]:
pet_df['breedname'].value_counts()[:15]

Domestic Short Hair       9240
Domestic Medium Hair       861
Domestic Long Hair         617
Bully Breed Mix            616
Labrador Retriever         258
Beagle                     242
Labrador Retriever/Mix     233
Chihuahua                  168
Hound/Mix                  152
Pitbull                    144
American                   137
German Shepherd Dog        130
Shep Mix                   110
Hound                      108
Mix                        106
Name: breedname, dtype: int64

In [26]:
pet_df['speciesname'].value_counts(normalize=True)

Cat                 0.600514
Dog                 0.345950
House Rabbit        0.018523
Guinea Pig          0.009315
Rat                 0.008726
Bird                0.004176
Livestock           0.002088
Hamster             0.001392
Mouse               0.001071
Hedgehog            0.000964
Gerbil              0.000964
Snake               0.000750
Ferret              0.000750
Wildlife - other    0.000750
Pig                 0.000696
Lizard              0.000642
Chicken             0.000535
Turtle              0.000428
Sugar Glider        0.000375
Opossum             0.000321
Fish                0.000321
Goat                0.000268
Tortoise            0.000161
Chinchilla          0.000161
Tarantula           0.000054
Raccoon             0.000054
Squirrel            0.000054
Name: speciesname, dtype: float64

Animals that arent dogs and cats account for less than 10% of our dataset so were going to drop them.

In [27]:
dog_cat_df = pet_df[(pet_df['speciesname'] == 'Dog') | (pet_df['speciesname'] == 'Cat')]

In [28]:
dog_cat_df['speciesname'].value_counts()

Cat    11217
Dog     6462
Name: speciesname, dtype: int64

In [29]:
dog_cat_df['animalage'].value_counts()

8 months.              1590
2 years 2 months.       477
1 year 2 months.        466
2 years 1 month.        452
1 year 1 month.         444
                       ... 
18 years 9 months.        1
0 days.                   1
31 years 10 months.       1
30 years 5 months.        1
28 years 2 months.        1
Name: animalage, Length: 315, dtype: int64

It is very unlikely that a cat or dog is a 99 or 30 years old, it's probably a typo, but it will be easier to drop outliers afer converting the age column to numeric datatype.

In [30]:
dog_cat_df['animalage'][dog_cat_df['animalage'].str.contains('month')].value_counts()

8 months.              1590
2 years 2 months.       477
1 year 2 months.        466
2 years 1 month.        452
1 year 1 month.         444
                       ... 
100 years 9 months.       1
33 years 0 months.        1
20 years 7 months.        1
100 years 0 months.       1
30 years 2 months.        1
Name: animalage, Length: 282, dtype: int64

### Make numeric age columns

##### Rules for the current age column
- If the animal is less than a week old the age is given in days
- So if the animal is less than 6 months old the age is given in weeks.
- If the animal is > 6 months and less than 1 year the age column only contains months
- If the animal is 1 or older the age column has years and months

In [31]:
#Get rid of the periods in the age column
dog_cat_df['animalage'] = dog_cat_df['animalage'].str.replace('.', '')

In [32]:
dog_cat_df['animalage'].value_counts()

8 months              1590
2 years 2 months       477
1 year 2 months        466
2 years 1 month        452
1 year 1 month         444
                      ... 
101 years 3 months       1
28 years 2 months        1
99 years 3 months        1
19 years 7 months        1
30 years 10 months       1
Name: animalage, Length: 315, dtype: int64

In [33]:
dog_cat_df['animalage'][dog_cat_df['animalage'].str.contains('day') == True].value_counts()

2 days    11
5 days     6
4 days     5
6 days     5
3 days     4
1 day      4
0 days     1
Name: animalage, dtype: int64

I want to make the age column uniform and then split it into 2 age columns, one for years and one for months

In [34]:
# convert all values that give the age in days to 1 week 
day_values = ['0 days', '1 day', '2 days', '3 days', '4 days', '5 days', '6 days']
dog_cat_df.loc[dog_cat_df['animalage'].isin(day_values), 'animalage'] = '1 week'

now convert our week values into months

In [35]:
# Round weeks to the nearest month
one_month = ['1 week', '2 weeks', '3 weeks', '4 weeks', '5 weeks', '6 weeks']
two_months = ['7 weeks', '8 weeks', '9 weeks', '10 weeks']
three_months = ['11 weeks', '12 weeks', '13 weeks', '14 weeks']
four_months = ['15 weeks', '16 weeks', '17 weeks', '18 weeks']
five_months = ['19 weeks', '20 weeks', '21 weeks', '22 weeks']
six_months = ['23 weeks', '24 weeks', '25 weeks', '26 weeks']

In [36]:
# if a value is in one of the lists above that value becomes 1 month, 2 months, etc
dog_cat_df.loc[dog_cat_df['animalage'].isin(one_month), 'animalage'] = '1 months'
dog_cat_df.loc[dog_cat_df['animalage'].isin(two_months), 'animalage'] = '2 months'
dog_cat_df.loc[dog_cat_df['animalage'].isin(three_months), 'animalage'] = '3 months'
dog_cat_df.loc[dog_cat_df['animalage'].isin(four_months), 'animalage'] = '4 months'
dog_cat_df.loc[dog_cat_df['animalage'].isin(five_months), 'animalage'] = '5 months'
dog_cat_df.loc[dog_cat_df['animalage'].isin(six_months), 'animalage'] = '6 months'

Round pets that are older than 6 months to 1 year

In [37]:
one_year = ['7 months', '8 months', '9 months', '10 months', '11 months']
dog_cat_df.loc[dog_cat_df['animalage'].isin(one_year), 'animalage'] = '1 year 0 months'


Create a column called months that records the ages of pets that are less than a year old.

In [38]:
dog_cat_df['age(months)'] = dog_cat_df['animalage'][(dog_cat_df['animalage'].str.contains('months') == True)\
                        & (~dog_cat_df['animalage'].str.contains('year') == True)].str.strip('months').astype(int)

Create a column that records the ages of pets that are older than 1 year.

In [39]:
dog_cat_df['age(years)'] = dog_cat_df['animalage'][(dog_cat_df['animalage'].str.contains('month') == True)\
                        & (dog_cat_df['animalage'].str.contains('year') == True)].str[:2].astype(int)

Drop rows that have dogs and cats older than 20, these pets typically dont live that long.

In [40]:
dog_cat_df.describe()

id  age(months)    age(years)
count  17679.000000  1277.000000  16402.000000
mean   63239.324170     4.097886      3.559810
std     4932.291626     1.873919      4.260322
min    15801.000000     1.000000      1.000000
25%    59430.000000     3.000000      1.000000
50%    63367.000000     4.000000      2.000000
75%    67338.500000     6.000000      4.000000
max    71756.000000     6.000000     99.000000

In [41]:
dog_cat_df['age(years)'] = dog_cat_df['age(years)'].fillna(0)

In [42]:
dog_cat_df = dog_cat_df[dog_cat_df['age(years)'] < 20]

In [43]:
dog_cat_df.describe()

id  age(months)    age(years)
count  17597.000000  1277.000000  17597.000000
mean   63246.267091     4.097886      3.145877
std     4934.419518     1.873919      3.274553
min    15801.000000     1.000000      0.000000
25%    59443.000000     3.000000      1.000000
50%    63372.000000     4.000000      2.000000
75%    67348.000000     6.000000      4.000000
max    71756.000000     6.000000     19.000000

### Clean breed name column

In [44]:
len(dog_cat_df['breedname'].value_counts())

960

960 different breeds

In [45]:
dog_cat_df['breedname'].value_counts()[:60]

Domestic Short Hair                   9193
Domestic Medium Hair                   859
Bully Breed Mix                        614
Domestic Long Hair                     613
Labrador Retriever                     256
Beagle                                 241
Labrador Retriever/Mix                 233
Chihuahua                              168
Hound/Mix                              151
Pitbull                                143
German Shepherd Dog                    130
Shep Mix                               110
Hound                                  108
Mix                                    105
Husky                                  100
Beagle/Mix                              76
Boxer/Mix                               76
Siamese                                 75
Border Collie/Mix                       66
Manx                                    52
Boxer                                   48
Border Collie                           47
Terrier                                 44
Great Pyren

In [46]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Rottweiler') == True].value_counts()

Labrador Retriever/Rottweiler     21
Rottweiler/Shep Mix               13
Rottweiler                        10
Rottweiler/Mix                     9
Hound/Rottweiler                   5
Rottweiler/Hound                   4
Australian Shepherd/Rottweiler     3
Rottweiler/Labrador Retriever      2
Doberman Pinscher/Rottweiler       2
Shepherd/Rottweiler                2
Rottweiler/Bully Breed Mix         2
Rottweiler/Boxer                   1
Saint Bernard/Rottweiler           1
Rottweiler/Great Pyrenees          1
Mastiff/Rottweiler                 1
Rottweiler/Chow Chow               1
Pitbull/Rottweiler                 1
Rottweiler/German Shepherd Dog     1
Beagle/Rottweiler                  1
Pug/Rottweiler                     1
German Shepherd Dog/Rottweiler     1
Name: breedname, dtype: int64

In [47]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Rottweiler') == True] = 'Rottweiler'

In [48]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Siberian Husky') == True].value_counts()

Siberian Husky/Hound            11
Siberian Husky                   9
Pomeranian/Siberian Husky        3
Husky/Siberian Husky             2
Shep Mix/Siberian Husky          2
Siberian Husky/Affenpinscher     1
Name: breedname, dtype: int64

There are a couple siberian husky mixes so I will just bin them all under 'siberian husky'

In [49]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Siberian Husky') == True] = 'Siberian Husky'

In [50]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Siamese') == True].value_counts()

Siamese                         75
Siamese/Domestic Short Hair     33
Domestic Short Hair/Siamese     20
Siamese/Mix                     16
Ragdoll/Siamese                  4
Siamese/Snowshoe                 4
Domestic Medium Hair/Siamese     3
Siamese/American Shorthair       3
Domestic Long Hair/Siamese       2
Siamese/Manx                     2
Snowshoe/Siamese                 2
Siamese/Balinese                 1
Siamese/DSH                      1
Siamese/Domestic Medium Hair     1
Siamese/Domestic Long Hair       1
Siamese/Egyptian Mau             1
Siamese/Ragdoll                  1
Name: breedname, dtype: int64

Same with Siamese cats

In [51]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Siamese') == True] = 'Siamese'

In [52]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Manx') == True].value_counts() 

Manx                         52
Manx/Domestic Short Hair     19
Domestic Short Hair/Manx      7
Manx/DSH                      4
Balinese/Manx                 1
Manx/Domestic Long Hair       1
Domestic Long Hair/Manx       1
Domestic Medium Hair/Manx     1
Manx/Mix                      1
Name: breedname, dtype: int64

In [53]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Manx') == True] = 'Manx'

In [54]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Domestic Long Hair') == True].value_counts()

Domestic Long Hair                        613
Domestic Long Hair/Maine Coon               5
Maine Coon/Domestic Long Hair               4
Ragamuffin/Domestic Long Hair               2
Domestic Long Hair/Persian                  2
Persian/Domestic Long Hair                  2
Domestic Long Hair/Domestic Short Hair      2
Name: breedname, dtype: int64

In [55]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Domestic Long Hair') == True] = 'Domestic Long Hair'

In [56]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Domestic Medium Hair') == True].value_counts()

Domestic Medium Hair               859
Domestic Medium Hair/Maine Coon      5
Balinese/Domestic Medium Hair        1
Name: breedname, dtype: int64

In [57]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Domestic Medium Hair') == True] = 'Domestic Medium Hair'

So i end up digging into the breeds and shortening the list of values in this column. A very long and tedious task

In [58]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Domestic Short Hair') == True] = 'Domestic Short Hair'

In [59]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Pitbull') == True] = 'Pitbull'

In [60]:
dog_cat_df['breedname'].value_counts()

Domestic Short Hair                       9218
Domestic Medium Hair                       865
Domestic Long Hair                         630
Bully Breed Mix                            614
Pitbull                                    276
                                          ... 
Anatolian Shepherd                           1
Anatolian Shepherd/German Shepherd Dog       1
Chinese Crested                              1
Beagle/Sheltie                               1
Dalmatian/Bully Breed Mix                    1
Name: breedname, Length: 867, dtype: int64

We still have 863 different breednames

In [61]:
# Repeat the same process to bin alot of the breedname values together 
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Labrador Retriever') == True] = 'Labrador Retriever'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Bully Breed') == True] = 'Bully Breed'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Beagle') == True] = 'Beagle'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Chihuahua') == True] = 'Chihuahua'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Hound') == True] = 'Hound'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('German Shepherd') == True] = 'German Shepherd'

In [62]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Husky') == True] = 'Husky'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Shep Mix') == True] = 'German Shepherd'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Boxer') == True] = 'Boxer'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Collie') == True] = 'Border Collie'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Pit Bull Terrier') == True] = 'Pitbull'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Terrier') == True] = 'Terrier'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Great Pyrenees') == True] = 'Great Pyrenees'

In [63]:
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Mastiff') == True] = 'Mastiff'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Golden Retriever') == True] = 'Golden Retriever'
dog_cat_df['breedname'][dog_cat_df['breedname'].str.contains('Retriever/Mix') == True] = 'Golden Retriever'

For the sake of the reader of this notebook I try to put this tedious process in the same cells

##### Split our dog and cat dataframe into a dog dataframe and a cat dataframe


In [64]:
# Create 2 sub dataframes, to clean dog and cat values individually 
cat_df = dog_cat_df[(dog_cat_df['speciesname'] == 'Cat')]
dog_df = dog_cat_df[(dog_cat_df['speciesname'] == 'Dog')]

### Clean Cat breednames and colors

In [65]:
# convert values with a count less than 10 into a bin called others
counts = cat_df['breedname'].value_counts()
cat_df['breedname'][cat_df['breedname'].isin(counts[counts < 10].index)] = 'Other'

In [66]:
cat_df['breedname'][cat_df['breedname'].str.contains('DSH') == True] = 'Domestic Short Hair'

#### Now we clean the color columns for the cats dataframe

In [67]:
cat_df['basecolour'].value_counts()[:20]

Black                     1813
Grey                      1031
Orange                     963
Black and White            911
Black and Brown            907
Brown and Black            587
Tortoiseshell              564
Grey and White             561
Orange and White           412
Calico                     392
Buff                       335
Torbie                     314
Black and grey             261
Brown, Black and White     240
Tabbico                    198
White and Black            172
Dilute tortoiseshell       162
Black, Brown and White     152
Dilute calico              146
Buff and white             127
Name: basecolour, dtype: int64

This column has different values for colors that are listed in different orders so I combine such color values.

In [68]:
cat_df['basecolour'][cat_df['basecolour'] == 'Black, Brown and White'] = 'Brown, Black and White'
cat_df['basecolour'][cat_df['basecolour'] == 'White and Black'] = 'Black and White'

In [69]:
cat_df['basecolour'][cat_df['basecolour'].str[:5] == 'White'].value_counts()

White               87
White and Grey      74
White and Orange    56
White and Tabby     48
White and Brown     27
White and Tan        8
White and Torti      1
Name: basecolour, dtype: int64

I combine the white colored cats into one value 

In [70]:
cat_df['basecolour'][cat_df['basecolour'].str[:5] == 'White'] = 'White'

In [71]:
len(cat_df['basecolour'].value_counts())

54

Now we have 54 color values for our cat df

In [72]:
cat_df['basecolour'].value_counts()

Black                        1813
Black and White              1083
Grey                         1031
Orange                        963
Black and Brown               907
Brown and Black               587
Tortoiseshell                 564
Grey and White                561
Orange and White              412
Brown, Black and White        392
Calico                        392
Buff                          335
Torbie                        314
White                         301
Black and grey                261
Tabbico                       198
Dilute tortoiseshell          162
Dilute calico                 146
Buff and white                127
Grey and black                 95
Grey Black and White           74
Lynx point                     53
Flame Point                    42
Brown                          36
Chocolate Point                35
Seal Point                     30
Silver                         29
Brown and White                27
Lilac Point                    21
Black and Grey

We have Grey and black and Black and grey values, so I will combine them

In [73]:
cat_df['basecolour'][cat_df['basecolour'] == 'Grey and black'] = 'Black and grey'

I decided to lowet the amount of values by binning every value that has a count less than 30 into a value called other

In [74]:
color_counts = cat_df['basecolour'].value_counts()

cat_df['basecolour'][cat_df['basecolour'].isin(color_counts[color_counts < 30].index)] = 'Other'

In [75]:
len(cat_df['basecolour'].value_counts())

26

In [76]:
cat_df['basecolour'].value_counts()

Black                     1813
Black and White           1083
Grey                      1031
Orange                     963
Black and Brown            907
Brown and Black            587
Tortoiseshell              564
Grey and White             561
Orange and White           412
Brown, Black and White     392
Calico                     392
Black and grey             356
Buff                       335
Torbie                     314
White                      301
Other                      250
Tabbico                    198
Dilute tortoiseshell       162
Dilute calico              146
Buff and white             127
Grey Black and White        74
Lynx point                  53
Flame Point                 42
Brown                       36
Chocolate Point             35
Seal Point                  30
Name: basecolour, dtype: int64

### Drop duplicated entries

##### We want to track the time between a pet entering the shelter for the first time until the time it is adopted so we will drop duplicates and keep the last entry.  Intakedate stays the same for duplicates so eventhough we only keep the last entry we still get hte original intake date.

In [77]:
cat_df['id'].duplicated().sum()

4060

In [78]:
# Drop duplicates keep last and the inplace arguement makes the change permanent to the dataframe
cat_df.drop_duplicates('id', keep='last', inplace=True)

#### Here we create the target column for our cat dataset

In [79]:
# Create the target column by subtracting the intake date from the movement date
cat_df['duration of stay'] = cat_df['movementdate'].dt.date - cat_df['intakedate'].dt.date

In [80]:
cat_df['duration of stay'].head()

0    2723 days
5    1964 days
14    163 days
17    122 days
26    151 days
Name: duration of stay, dtype: timedelta64[ns]

### Clean Dog dataframe

In [81]:
dog_df['basecolour'].value_counts()

Black                        786
Black and White              650
Tricolour                    411
Black and Tan                360
Tan                          357
Brindle                      281
Red                          277
Tan and White                252
Brown                        251
White                        251
Brown and White              237
Black and Brown              209
White and Black              179
White and Brown              169
Brown and Black              147
Cream                        144
Brindle and White            141
White and Tan                132
Chocolate                    125
Grey and White               109
Red and White                 93
Golden                        92
Blue                          74
Grey                          69
Yellow                        57
Blue merle                    47
Tan and Black                 40
Fawn                          39
Black merle                   35
Brown, Black and White        35
White and 

The dataset has different values for 'grey and white' and 'white and grey', so I will combint them 

In [82]:
dog_df['basecolour'][dog_df['basecolour'] == 'White and Grey'] = 'Grey and White'

Yellow and fawn dogs are very similar so I will combine the 2 values under the fawn color

In [83]:
dog_df['basecolour'][dog_df['basecolour']== 'Yellow'] = 'Fawn'

Chocolate and Brown dogs look very similar so I will combine the 2 classes

In [84]:
dog_df['basecolour'][dog_df['basecolour']== 'Chocolate'] = 'Brown'

Combine 'black and brown' values with 'brown and black' values and do the same with dogs considered white and black.

In [85]:
dog_df['basecolour'][dog_df['basecolour'].str.contains('White')].value_counts()

Black and White           650
Tan and White             252
White                     251
Brown and White           237
White and Black           179
White and Brown           169
Brindle and White         141
Grey and White            139
White and Tan             132
Red and White              93
Brown, Black and White     35
White and Brindle          34
Black, Brown and White     29
White and Liver            22
Orange and White            9
Liver and White             5
White and Orange            2
Grey Black and White        1
Name: basecolour, dtype: int64

Many color values just written in different orders.

In [86]:
dog_df['basecolour'][dog_df['basecolour']== 'White and Black'] = 'Black and White'
dog_df['basecolour'][dog_df['basecolour']== 'Brown, Black and White'] = 'Black, Brown and White'
dog_df['basecolour'][dog_df['basecolour']== 'White and Brown'] = 'Brown and White'
dog_df['basecolour'][dog_df['basecolour']== 'Tan and White'] = 'White and Tan'
dog_df['basecolour'][dog_df['basecolour']== 'Brown and Black'] = 'Black and Brown'


In [87]:
dog_df['basecolour'].value_counts()

Black and White              829
Black                        786
Tricolour                    411
Brown and White              406
White and Tan                384
Brown                        376
Black and Tan                360
Tan                          357
Black and Brown              356
Brindle                      281
Red                          277
White                        251
Cream                        144
Brindle and White            141
Grey and White               139
Fawn                          96
Red and White                 93
Golden                        92
Blue                          74
Grey                          69
Black, Brown and White        64
Blue merle                    47
Tan and Black                 40
Black merle                   35
White and Brindle             34
Black and grey                28
Brindle and Black             25
Red merle                     24
Tan and Brown                 23
Black and Brindle             22
White and 

In [88]:
dog_df['basecolour'][dog_df['basecolour']== 'Grey and black'] = 'Black and grey'
dog_df['basecolour'][dog_df['basecolour']== 'Tan and Black'] = 'Black and Tan'

In [89]:
# bin all color values with a count less than 30 into a value called other
dog_color_counts = dog_df['basecolour'].value_counts()

dog_df['basecolour'][dog_df['basecolour'].isin(dog_color_counts[dog_color_counts < 30].index)] = 'Other'

In [90]:
# Drop duplicates and keep last just like we did with the cat df
dog_df.drop_duplicates('id', keep='last', inplace=True)

In [91]:
dog_df.duplicated().sum()

0

##### Dog Breednames

In [92]:
dog_df['breedname'].value_counts()[:40]

Labrador Retriever         708
Bully Breed                544
Beagle                     380
Terrier                    363
Hound                      316
German Shepherd            290
Chihuahua                  236
Pitbull                    207
Husky                      141
Border Collie              115
Boxer                       90
Golden Retriever            65
Mix                         65
Great Pyrenees              53
Rottweiler                  52
Mastiff                     31
Shih Tzu                    31
Coonhound                   26
Australian Shepherd         24
Heeler                      22
Miniature Pinscher          21
Mountain Cur                19
Pomeranian                  18
Pug                         17
Cattle Dog                  15
Dachshund                   14
Heeler/Mix                  14
Australian Shepherd/Mix     13
Cocker Spaniel              12
American Bulldog            12
Great Dane                  11
Poodle                      10
Spaniel 

#### We have 200 values that have counts that are less than 5 so we will bin these into a value called 'other'

In [93]:
dog_df['breedname'].value_counts(ascending =True)[:200]

Italian Greyhound/Whippet              1
Crossbreed                             1
Schnauzer, Miniature/Cocker Spaniel    1
Cavalier King Charles Spaniel/Mix      1
English Bulldog/Shar Pei               1
                                      ..
Schnauzer, Standard                    4
Lhasa Apso/Poodle                      4
Maltese/Poodle                         4
Foxhound                               4
Bernese Mountain Dog                   4
Name: breedname, Length: 200, dtype: int64

In [94]:
dog_breed_counts = dog_df['breedname'].value_counts()

dog_df['breedname'][dog_df['breedname'].isin(dog_breed_counts[dog_breed_counts < 5].index)] = 'Other'

In [95]:
dog_df['breedname'].value_counts()

Labrador Retriever    708
Bully Breed           544
Beagle                380
Terrier               363
Hound                 316
                     ... 
Pointer/Mix             5
Retriever               5
Pomeranian/Mix          5
Carolina Dog/Mix        5
Saint Bernard           5
Name: breedname, Length: 67, dtype: int64

In [96]:
dog_df['breedname'][dog_df['breedname'].str.contains('Coonhound') == True] = 'Coonhound'
dog_df['breedname'][dog_df['breedname'].str.contains('American Bulldog') == True] = 'American Bulldog'
dog_df['breedname'][dog_df['breedname'].str.contains('Dachshund') == True] = 'Dachshund'
dog_df['breedname'][dog_df['breedname'].str.contains('Cattle Dog') == True] = 'Cattle Dog'
dog_df['breedname'][dog_df['breedname'].str.contains('Shiba Inu') == True] = 'Shiba Inu'
dog_df['breedname'][dog_df['breedname'].str.contains('Australian Shepherd') == True] = 'Australian Shepherd'
dog_df['breedname'][dog_df['breedname'].str.contains('Schnauzer') == True] = 'Schnauzer'
dog_df['breedname'][dog_df['breedname'].str.contains('Foxhound') == True] = 'Foxhound'

In [97]:
dog_df['breedname'][dog_df['breedname'].str.contains('Pomeranian') == True] = 'Pomeranian'
dog_df['breedname'][dog_df['breedname'].str[:7] == 'Spaniel'] = 'Spaniel'
dog_df['breedname'][dog_df['breedname'].str.contains('Heeler') == True] = 'Heeler'
dog_df['breedname'][dog_df['breedname'].str[:6] == 'Poodle'] = 'Poodle'

In [98]:
len(dog_df['breedname'].value_counts())

50

Now we are down to 49 different breedname values

In [99]:
dog_df.head()

id intakedate  intakereason  istransfer identichipnumber animalname  \
1   15932 2009-12-08        Moving       False       0A11675477      Gonzo   
3   28859 2012-08-10     Abandoned       False       0A13253C7B     Maggie   
7   31469 2013-03-26  Incompatible       False  981020007006095     Bonnie   
12  46238 2016-10-09     Abandoned       False  981020019070093  Tennessee   
15  46693 2016-11-21     Abandoned       False  981020017897009       Baby   

          breedname       basecolour speciesname          animalage sexname  \
1   German Shepherd              Tan         Dog  10 years 9 months    Male   
3             Husky            Other         Dog    8 years 1 month  Female   
7             Other  Brown and White         Dog   7 years 6 months  Female   
12           Heeler  Black and Brown         Dog   5 years 7 months    Male   
15          Pitbull            White         Dog   5 years 9 months    Male   

              location movementdate movementtype returndate  puttosleep  \
1       Adoptable Dogs   2017-04-24     Adoption        NaN       False   
3   Canine intake room   2020-02-29    Reclaimed        NaN       False   
7      Stray Dogs Beta   2017-05-09     Adoption        NaN       False   
12   Special Care Dogs   2018-02-14     Adoption        NaN       False   
15      Adoptable Dogs   2017-04-07     Adoption        NaN       False   

    age(months)  age(years)  
1           NaN        10.0  
3           NaN         8.0  
7           NaN         7.0  
12          NaN         5.0  
15          NaN         5.0

### Create Target Column 

In [100]:
# Create target column the same way that we did for the cat dataframe
dog_df['duration of stay'] = dog_df['movementdate'].dt.date - dog_df['intakedate'].dt.date

In [101]:
dog_df.head()

id intakedate  intakereason  istransfer identichipnumber animalname  \
1   15932 2009-12-08        Moving       False       0A11675477      Gonzo   
3   28859 2012-08-10     Abandoned       False       0A13253C7B     Maggie   
7   31469 2013-03-26  Incompatible       False  981020007006095     Bonnie   
12  46238 2016-10-09     Abandoned       False  981020019070093  Tennessee   
15  46693 2016-11-21     Abandoned       False  981020017897009       Baby   

          breedname       basecolour speciesname          animalage sexname  \
1   German Shepherd              Tan         Dog  10 years 9 months    Male   
3             Husky            Other         Dog    8 years 1 month  Female   
7             Other  Brown and White         Dog   7 years 6 months  Female   
12           Heeler  Black and Brown         Dog   5 years 7 months    Male   
15          Pitbull            White         Dog   5 years 9 months    Male   

              location movementdate movementtype returndate  puttosleep  \
1       Adoptable Dogs   2017-04-24     Adoption        NaN       False   
3   Canine intake room   2020-02-29    Reclaimed        NaN       False   
7      Stray Dogs Beta   2017-05-09     Adoption        NaN       False   
12   Special Care Dogs   2018-02-14     Adoption        NaN       False   
15      Adoptable Dogs   2017-04-07     Adoption        NaN       False   

    age(months)  age(years) duration of stay  
1           NaN        10.0        2694 days  
3           NaN         8.0        2759 days  
7           NaN         7.0        1505 days  
12          NaN         5.0         493 days  
15          NaN         5.0         137 days

Oue target column is currently a timedelta datatype, changing it to a numeric datatype will make it easier to work with. 

In [102]:
dog_df['duration of stay'] = dog_df['duration of stay'].dt.days

In [103]:
dog_df['duration of stay'].describe()

count    4474.000000
mean       51.284980
std       144.016508
min        -6.000000
25%         3.000000
50%        11.000000
75%        32.000000
max      2759.000000
Name: duration of stay, dtype: float64

We have values that are less than 0 so, I check how many values there are.

In [104]:
dog_df['duration of stay'][dog_df['duration of stay'] < 0].value_counts()

-6.0    1
Name: duration of stay, dtype: int64

A pet cant be in a shelter negative 1 days so we will drop that row.

In [105]:
dog_df = dog_df[dog_df['duration of stay'] >= 0]

In [106]:
dog_df['duration of stay'].describe()

count    4473.000000
mean       51.297787
std       144.030062
min         0.000000
25%         3.000000
50%        11.000000
75%        32.000000
max      2759.000000
Name: duration of stay, dtype: float64

In [107]:
dog_df['duration of stay'][dog_df['duration of stay'] > 1500].value_counts()

1505.0    1
2759.0    1
2694.0    1
Name: duration of stay, dtype: int64

Drop some outliers in order to make the model more accurate.

In [108]:
dog_df = dog_df[dog_df['duration of stay'] < 1500]

### cat_df Create Target

In [109]:
cat_df.head()

id intakedate intakereason  istransfer identichipnumber   animalname  \
0   15801 2009-11-28       Moving       False       0A115D7358       Jadzia   
5   30812 2013-01-11    Abandoned       False       0A13403D4D  Pretty Girl   
14  46437 2016-10-26    Abandoned       False  981020017650993         Nova   
17  47120 2017-01-11    Abandoned       False  981020021072899       Vivian   
26  47198 2017-01-22    Abandoned       False  981020021020775     Mama Mia   

              breedname            basecolour speciesname          animalage  \
0   Domestic Short Hair         Tortoiseshell         Cat  11 years 8 months   
5   Domestic Short Hair  Dilute tortoiseshell         Cat  10 years 6 months   
14   Domestic Long Hair                 Black         Cat  12 years 3 months   
17                Other                 Other         Cat   6 years 5 months   
26  Domestic Short Hair        Grey and White         Cat  4 years 10 months   

   sexname                    location movementdate movementtype  \
0   Female  Adoptable Cat Glass Colony   2017-05-13     Adoption   
5   Female                  Stray Cats   2018-05-29     Adoption   
14  Female                  Stray Cats   2017-04-07     Adoption   
17  Female                      Foster   2017-05-13     Adoption   
26  Female  Adoptable Cat Glass Colony   2017-06-22     Adoption   

                returndate  puttosleep  age(months)  age(years)  \
0                      NaN       False          NaN        11.0   
5                      NaN       False          NaN        10.0   
14  02/09/2018 12:00:00 AM        True          NaN        12.0   
17                     NaN       False          NaN         6.0   
26                     NaN       False          NaN         4.0   

   duration of stay  
0         2723 days  
5         1964 days  
14         163 days  
17         122 days  
26         151 days

In [110]:
cat_df['duration of stay'].dt.days.head()

0     2723.0
5     1964.0
14     163.0
17     122.0
26     151.0
Name: duration of stay, dtype: float64

In [111]:
# make the target in this dataframe numeric
cat_df['duration of stay'] = cat_df['duration of stay'].dt.days

In [112]:
cat_df['duration of stay'].describe()

count    7075.000000
mean       48.791943
std       100.561862
min        -4.000000
25%        11.000000
50%        29.000000
75%        55.000000
max      2723.000000
Name: duration of stay, dtype: float64

In [113]:
cat_df['duration of stay'][cat_df['duration of stay'] > 1500].value_counts()

1964.0    1
2723.0    1
Name: duration of stay, dtype: int64

Drop the outliers and drop the row with the cat that stayed in the shelter -4 days.

In [114]:
cat_df = cat_df[cat_df['duration of stay'] < 1500]

In [115]:
cat_df[cat_df['duration of stay'] < 0]

id          intakedate intakereason  istransfer identichipnumber  \
5055  59647 2018-05-25 17:53:21        other       False              NaN   

     animalname            breedname basecolour speciesname animalage  \
5055   Garbanzo  Domestic Short Hair      Black         Cat  1 months   

      sexname            location movementdate movementtype  \
5055  Unknown  Canine intake room   2018-05-21       Foster   

                  returndate  puttosleep  age(months)  age(years)  \
5055  06/05/2018 12:00:00 AM       False          1.0         0.0   

      duration of stay  
5055              -4.0

In [116]:
cat_df = cat_df[cat_df['duration of stay'] >= 0]

In [117]:
cat_df['duration of stay'].describe()

count    7072.000000
mean       48.150452
std        92.661009
min         0.000000
25%        11.000000
50%        28.500000
75%        55.000000
max      1478.000000
Name: duration of stay, dtype: float64

#### Saved these csvs to use for modeling in another notebook

In [116]:
# cat_df.to_csv('cat_df.csv')
# dog_df.to_csv('dog_df.csv')